scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,882 @@
+            { 192, 155.500, -0.250, -30.000, "0x00006D0500000000000000000000000000000000", },
+            {  64, -46.800, -4.916, 227.570, "0x00006D0500000000000000000000000000000000", },
+            {  64, -33.200, -4.982, 227.570, "0x00006D0500000000000000000000000000000000", },
+        },
+        helpers =
+        {
+            { 192, 165.000, -0.250, -28.800, "0x01001E0582117020173017401750006000700000", },
+            { 192, 164.000, -0.250, -28.800, "0x01001E0682117020173017401750006000700000", },
+            { 192, 155.500, -0.250, -28.800, "0x01001E0782117020173017401750006000700000", },
+            {  63, -46.800, -4.900, 226.220, "0x01001E0882117020173017401750006000700000", },
+            {  63, -33.200, -4.900, 226.220, "0x01001E0182117020173017401750006000700000", },
+        },
+    },
+
+    [xi.zone.WINDURST_WOODS] =
+    {
+        cs = 785,
+        decorations =
+        {
+            { 128,  107.630, -5.000, -33.200, "0x00006D0500000000000000000000000000000000", },
+            { 128,  107.630, -5.000, -46.800, "0x00006D0500000000000000000000000000000000", },
+            {   0, -107.000, -5.250,  35.500, "0x00006D0500000000000000000000000000000000", },
+            {   0, -107.000, -5.250,  44.500, "0x00006D0500000000000000000000000000000000", },
+        },
+        helpers =
+        {
+            {   0, -105.500, -5.250,  35.500, "0x01001E0282117020173017401750006000700000", },
+            {   0, -105.500, -5.250,  44.500, "0x01001E0782117020173017401750006000700000", },
+            { 127,  106.500, -5.000, -33.200, "0x01001E0382117020173017401750006000700000", },
+            { 127,  106.500, -5.000, -46.800, "0x01001E0482117020173017401750006000700000", },
+            { 127, -155.000, -4.000, -27.000, "0x01001E0100107120173017401750006000700000", },
+            { 127, -155.000, -4.000, -33.000, "0x01001E0282117120173017401750006000700000", },
+        },
+    },
+}
+
+-- ----------------------------------
+-- Helpers
+-- ----------------------------------
+
+xi.events.egg_hunt.charToEgg = function(char)
+    local letter = string.byte(string.lower(char)) - 97
+    return xi.items.A_EGG + letter
+end
+
+xi.events.egg_hunt.eggInList = function(eggs, egg)
+    for j = 1, #eggs do
+        if eggs[j] and eggs[j][1] == egg then
+            return j
+        end
+    end
+
+    return -1
+end
+
+xi.events.egg_hunt.stringToEggs = function(text)
+    local str = string.lower(text)
+    local eggs = {}
+
+    for i = 1, #str do
+        local letter = string.byte(string.sub(str, i, i))
+        local egg = xi.items.A_EGG + (letter - 97)
+        local pos = xi.events.egg_hunt.eggInList(eggs, egg)
+
+        if pos > -1 then
+            eggs[pos][2] = eggs[pos][2] + 1
+        else
+            table.insert(eggs, { egg, 1 })
+        end
+    end
+
+    return eggs
+end
+
+local hasFirstThree = function(player)
+    return (
+        player:getVar(settings.VAR.FIRST_THREE) == 1 or
```

Comment:
Minor, but, we use getCharVar to fetch player variables.

In quest scripts we don't, but outside of them we do, for consistency.